### PR TITLE
606 pyfunc reload fails

### LINKED ISSUE #606: PyFunc reload fails
Whenever I modify the PyFunc, it can't reload and I have to restart the module.

```python
 INFO  insight::savant::base::pyfunc                        > Dev mode is enabled and changes in "src.module.custom_pyfunc.HeightCalcPyFunc" are detected; reloading.
 INFO  insight::savant::base::pyfunc                        > The module "src.module.custom_pyfunc.HeightCalcPyFunc" reloading complete: OK
 INFO  insight::savant::base::pyfunc                        > Dev mode is enabled and changes in "src.module.custom_pyfunc.ObjectTrackerPyFunc" are detected; reloading.
 ERROR insight::savant::base::pyfunc                        > Error while reloading module instance.
Aliases for entries in sys.path:
    <distpkg>: /usr/local/lib/python3.8/dist-packages
    <py>     : /usr/lib/python3.8
Traceback (most recent call last):
    <distpkg> /savant/base/pyfunc.py:199  load_user_code  module_instance = reload_module(self._module_instance)
    <distpkg> /savant/base/pyfunc.py:323  reload_module   return reload(module)
    <py> /importlib/__init__.py:148       reload          raise ImportError(msg.format(name), name=name)
ImportError: module src.module.custom_pyfunc not in sys.modules
 INFO  insight::savant::base::pyfunc                        > The module "src.module.custom_pyfunc.ObjectTrackerPyFunc" reloading finished: Fail
Aliases for entries in sys.path:
    <distpkg>: /usr/local/lib/python3.8/dist-packages
    <p0>     : /opt/savant/gst_plugins/python
```

### PR DESCRIPTION
**Pull request by dorgun**

### CODE REVIEW

File: savant/utils/modules_cache.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    def get_module_instance(self, module_spec: ModuleSpec) -> Optional[ModuleType]:
+        """Get module instance by module specification.
+
+        :param module_spec: user module specification.
+        :return: module instance.
+        """
+        module_file = module_spec.origin
+        if module_file not in self._cache_module:
+            logger.debug(f'{module_file} is not cached, importing.')
+            if module_spec is None:
```
Comment (denisvmedyantsev) [inline savant/utils/modules_cache.py]: Module spec is required, the condition doesn't work here.

**Review by denisvmedyantsev — CHANGES_REQUESTED** [review]

File: savant/utils/modules_cache.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        self._cache_module: Dict[str, ModuleType] = {}
+
+    def get_module_instance(self, module_spec: ModuleSpec) -> Optional[ModuleType]:
+        """Get module instance by module specification.
+
+        :param module_spec: user module specification.
+        :return: module instance.
+        """
+        module_file = module_spec.origin
+        if module_file not in self._cache_module:
```
Comment (abramov-oleg) [inline savant/utils/modules_cache.py]: It is possible for `origin` to be None, I think it is better to explicitly handle this case.

File: savant/utils/modules_cache.py (hunk 1 above)
Comment (dorgun) [reply savant/utils/modules_cache.py]: updated

File: savant/utils/modules_cache.py (hunk 2 above)
Comment (dorgun) [reply savant/utils/modules_cache.py]: updated

**Review by denisvmedyantsev — APPROVED** [review]

**Pull request merged**